2048/2048/src/main/java/com/appcoins/eskills2048/model/RoomApiMapper.java

```diff
@@ -0,0 +1,33 @@
+package com.appcoins.eskills2048.model;
+
+import androidx.annotation.NonNull;
+import java.util.Objects;
+import org.json.JSONObject;
+import retrofit2.HttpException;
+
+public class RoomApiMapper {
+
+  public RoomResponseErrorCode mapException(@NonNull Throwable throwable){
+    String body;
+    RoomResponseErrorCode response;
+    if(throwable instanceof HttpException) {
+      HttpException exception = (HttpException) throwable;
+      try {
+        body = Objects.requireNonNull(Objects.requireNonNull(exception.response())
```

Comment:
why don't you use Gson to parse the response?

Reply:
Changed it!